scvi/inference/trainer.py

```diff
@@ -153,13 +161,14 @@ def train(
         ):
             self.on_epoch_begin()
             for tensors_list in self.data_loaders_loop():
+                self.on_iteration_begin()
                 if tensors_list[0][0].shape[0] < 3:
                     continue
-                loss = self.loss(*tensors_list)
-                self.check_training_status(loss, max_nans)
+                self.current_loss = loss = self.loss(*tensors_list)
```

Comment:
what's the implication of the training status now being checked AFTER the backward call?

Reply:
I don't think it has any impact: 

```
import torch

param = torch.tensor([10.], requires_grad=True)
optim = torch.optim.SGD([param], lr=0.1)
loss = param ** 2
print(loss)
loss.backward()
print(loss)

```

Reply:
Sorry got confused with the iteration end and the epoch end and the early stopping stuff.